OCPBUGS-11767 unstable cluster due to admission webhooks (#104913)
openshift/openshift-docs: modules/preventing-cluster-webhook-failure.adoc

// Module included in the following assemblies:
//
// * builds/triggering-builds-build-hooks.adoc

:_mod-docs-content-type: CONCEPT
[id="third-party-cluster-webhook-failure_{context}"]
= Prevent cluster failure due to webhooks

[role="_abstract"]

> To prevent potential cluster failure and ensure pods can always start, you must configure third-party admission webhooks to exclude infrastructure namespaces. Implementing specific selectors and adopting `ValidatingAdmissionPolicy` provides a more stable environment for cluster recovery and management.

Reviewer: missing an article?
```suggestion
To prevent potential cluster failure and ensure pods can always start, you must configure third-party admission webhooks to exclude infrastructure namespaces. Implementing specific selectors and adopting a `ValidatingAdmissionPolicy` provides a more stable environment for cluster recovery and management.
```

> When possible, use `ValidatingAdmissionPolicy` instead of webhooks. It does not require an external service, has no timeout limitations, and cannot cause cluster-wide failures.

Reviewer 2: Bryce or Kevin can confirm this, but my understanding is that validating admission policy is type of webhook. I think the issue might be that mutating and/or third party admission webhooks are the bit that need the config and VAP are a kind of webhook that handles it for you. But I might be wrong. :)

Nit: remember to use a noun after a specific API or resource reference.

If you use admission webhooks take the following precautions:

> * Configure the webhook to exclude OpenShift and Kubernetes infrastructure namespaces.

Reviewer: ```suggestion
* Configure the webhook to exclude {product-title} and Kubernetes infrastructure namespaces.
```

* Configure webhook timeouts to 10 seconds or less to provide a safety buffer for the system enforced 13-second limit.

> * Set `failurePolicy: Ignore` for non-critical webhooks so that requests can proceed if the webhook is unavailable.

Reviewer: re: backticked items as nouns, consider:
```suggestion
* Set the `failurePolicy` value to `Ignore` for non-critical webhooks so that requests can proceed if the webhook is unavailable.
```
